### PR TITLE
Implement ZipExtFile in the zipfile type stubs

### LINKED ISSUE #1224: Missing name ZipExtFile in the stub for the zipfile module
Hello,

My env is Win7P+SP1 x64, Py 3.5.2 32b, Mypy 0.501.

When I do this
`from zipfile import ZipExtFile`

mypy returns this error

`file_compression.py:38: error: Module 'zipfile' has no attribute 'ZipExtFile'`

I was told in mypy gitter that the problem is a missing name in the stub for the zipfile module.

Best regards,

JM

### PR DESCRIPTION
**Pull request by CraftSpider**
Adds the ZipExtFile as a stub in zipfile.pyi. Closes #1224

Known issues:
- [X] read1 errors due to incompatible superclass type, however this is how the implementation is written
- [X] python2 errors with `name` defined differently in superclass, however this is the type actually provided by the implementation

~~Assistance for the proper resolution of these two issues is welcome.~~

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
For `read1`, you'll just have to `# type: ignore`; that's what we do in a few other places where the real world gives us incompatible overrides.

For `name`, it looks like it can be both str and unicode in Python 2, but defaults to str. It may be easier to just use `name: str`.

**Comment by CraftSpider** [comment]
Alright, thank you. I'll update the PR with those changes

**Comment by CraftSpider** [comment]
Any updates on this? I understand if people are busy, but also I want to make sure this doesn't fall completely by the wayside

**Comment by JelleZijlstra** [comment]
Looks like I started a review but never submitted it, sorry for that!

File: stdlib/2and3/zipfile.pyi [hunk 1]
```diff
@@ -23,6 +24,27 @@ error = BadZipfile
...
+
+    def __init__(self, fileobj: IO[bytes], mode: str, zipinfo: ZipInfo, decrypter: Optional[Callable[[Sequence[int]], bytes]] = ..., close_fileobj: bool = ...) -> None: ...
+    def __repr__(self) -> str: ...
+    def readline(self, limit: int = ...) -> bytes: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def readable(self) -> bool: ...
+    def read(self, n: Optional[int] = ...) -> bytes: ...
+    def read1(self, n: int) -> bytes: ...
+    def close(self) -> None: ...
+    def seekable(self) -> bool: ...
```
[the remark this answers is not in the document]
Comment (CraftSpider) [reply stdlib/2and3/zipfile.pyi]: Removed redundant overrides

**Comment by CraftSpider** [comment]
Thanks for the patience, had stuff that took priority over the weekend. Fixed the noted issues

File: stdlib/2and3/zipfile.pyi [hunk 2]
```diff
@@ -23,6 +24,25 @@ error = BadZipfile
...
+        MAX_SEEK_READ: int = ...
+
+    newlines: Optional[List[bytes]]
+    mode: str
+    name: str
+
+    def __init__(self, fileobj: IO[bytes], mode: str, zipinfo: ZipInfo, decrypter: Optional[Callable[[Sequence[int]], bytes]] = ..., close_fileobj: bool = ...) -> None: ...
+    def __repr__(self) -> str: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def read1(self, n: int) -> bytes: ...  # type: ignore
```
Comment (JelleZijlstra) [inline stdlib/2and3/zipfile.pyi]: The argument is actually `Optional[int]` since the implementation checks for None.

**Comment by CraftSpider** [comment]
Any updates on the review for this? I'm not sure if there's anything left for me to do

**Comment by JelleZijlstra** [comment]
Sorry for the delay!

**Pull request merged**